tests/raft_log_replication.py

```diff
@@ -593,7 +593,419 @@ async def _unittest_raft_log_replication() -> None:
+    assert raft_node_2._log[2].term == 5
+    assert raft_node_2._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_2._log[2].entry.value == 8
+    assert raft_node_2._log[3].term == 6
+    assert raft_node_2._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_2._log[3].entry.value == 9
+    assert raft_node_2._commit_index == 3
+
+    assert len(raft_node_3._log) == 1 + 3
+    assert raft_node_3._log[0].term == 0
+    assert raft_node_3._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_3._log[0].entry.value == 0
+    assert raft_node_3._log[1].term == 4
+    assert raft_node_3._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_3._log[1].entry.value == 7
+    assert raft_node_3._log[2].term == 5
+    assert raft_node_3._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_3._log[2].entry.value == 8
+    assert raft_node_3._log[3].term == 6
+    assert raft_node_3._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_3._log[3].entry.value == 9
+    assert raft_node_3._commit_index == 3
+
+    _logger.info("================== TEST 2: Append new log entry fron another LEADER ==================")
+
+    new_entry = sirius_cyber_corp.LogEntry_1(
+        term=7,
+        entry=sirius_cyber_corp.Entry_1(
+            name=uavcan.primitive.String_1(value="top_4"),
+            value=13,
+        ),
+    )
+    request = sirius_cyber_corp.AppendEntries_1.Request(
+        term=raft_node_2._term,
+        prev_log_index=3,  # index of top_3
+        prev_log_term=raft_node_2._log[3].term,
+        log_entry=new_entry,
+    )
+    metadata = pycyphal.presentation.ServiceRequestMetadata(
+        client_node_id=43,
+        timestamp=time.time(),
+        priority=0,
+        transfer_id=0,
+    )
+
+    response = await raft_node_2._serve_append_entries(request, metadata)
+
+    assert response.success == True
+
+    await asyncio.sleep(TERM_TIMEOUT + 1)
+
+    assert len(raft_node_2._log) == 1 + 4
+    assert raft_node_2._log[0].term == 0
+    assert raft_node_2._log[0].entry.value == 0
+    assert raft_node_2._log[1].term == 4
+    assert raft_node_2._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_2._log[1].entry.value == 7
+    assert raft_node_2._log[2].term == 5
+    assert raft_node_2._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_2._log[2].entry.value == 8
+    assert raft_node_2._log[3].term == 6
+    assert raft_node_2._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_2._log[3].entry.value == 9
+    assert raft_node_2._log[4].term == 7
+    assert raft_node_2._log[4].entry.name.value.tobytes().decode("utf-8") == "top_4"
+    assert raft_node_2._log[4].entry.value == 13
+    assert raft_node_2._commit_index == 4
```

Comment:
From the comment above the test I was under the impression it would result in the following:

```
   Step 3: Replace Log Entry 3 with a New Entry fom LEADER Node 42
   ____________
  | 3          |     Log index
  | 7          |     Log term
  | top_3 <= 10|     Name <= value
  |____________|

  Result:
   ____________ ____________ ____________ _____________
  | 0          | 1          | 2          | 3           |     Log index
  | 0          | 4          | 5          | 7           |     Log term
  | empty <= 0 | top_1 <= 7 | top_2 <= 8 | top_3 <= 10 |     Name <= value
  |____________|____________|____________|_____________|
```